### DISCUSSION TITLE
FoxESS AIO-H1-5.0

### DISCUSSION THREAD

**Discussion by bperemans** [StealthChesnut/HA-FoxESS-Modbus#18] [Show and tell]
I can confirm that it is compatible with a AIO-H1-5.0 inverter/battery.
Thanks for this great HA integration!

**Comment by StealthChesnut** [comment]
Thanks for the report @bperemans ! 

Can you share which method you used please - was it via a USB device or via a WiFi device?

**Reply by Parisn0** [reply]
Mine is Inverter Model : H1-5.0-E and works perfectly with a WiFi device.

Was reading through this to see if there are any plans to add the force charge? FoxESS website occasionally goes down and this is most frustrating as I cannot change the force charge settings and either waste money charging on sunny days or do not charge at Octopus Go low rate on low generation days.

**Reply by d-p-simpson** [reply]
I have the same.  I'm currently using a script to enable charging using the website API.  However, as you say, when the website is down, this doesn't work.

Unfortunately, the charging isn't exposed on the LAN or RS485.
I've tried to do MITM on the WiFi dongle, and can see traffic to the fox website on port 10001.  Encoding seems to be UINT16, but is obfuscated somehow.   The LAN adaptor only does one HTTP call to the internet when the inveter starters, and that appears to be just a metadata pull for firmware versions.

My only remaining options are to to build a tap onto the WiFi adaptor (4 pins).  I'm hoping that these 4 pins aren't in fact USB and are x2 power for the WiFi and x2 RS485.  If it is RS485, it will be possible to see what data the WiFi adaptor sends into the inverter to enable charging.

Failing that, there is the COM port which has external debug which might assist somehow.  That would only be a last resort however.

**Reply by StealthChesnut** [reply]
@djdeant the wifi dongle tap has been tried here: https://github.com/macxq/foxess-ha/discussions/49

I think they could only monitor, and even then the data isn't easy to read.

**Reply by d-p-simpson** [reply]
@StealthChesnut  Thanks!  I might give the COM a try then possibly

**Reply by Parisn0** [reply]
It is rather poor that we are limited to FoxCloud or the settings on the inverter to set force charging, mine is not too bad in that the inverter is on a wall in my garage so I can set force charge in a few min, they were originally putting the inverter and batteries in the loft - much more problematic. I expect at some point they will start charging a monthly fee for access to FoxCloud.

**Reply by StealthChesnut** [reply]
@djdeant you can access modbus information over the LAN port. I've put it into this integration this week. Think there's still a couple of bugs to iron out with the HACS install, but it's getting there.

It might be possible to change settings this way, but I'll need to find where they are first in the address space. Still no documentation for these inverters.

**Reply by bkortleven** [reply]
@d-p-simpson : did you find out how the 4-pin is wired? USB or RS485+power?
as the v1.56 comm update bricked the ethernet port (or disabled it on purpose) I'm looking for (yet) a(nother) alternative way of getting the data out of the inverter to use it in HA as I have been doing for months now...
If you have time, please elaborate on the setup you managed to get working...! Thanks!

**Comment by bperemans** [comment]
﻿It was via USB. I bought the DSD Tech SH-U10 Usb to Rs485 converter.

**Comment by Daibutt** [comment]
https://foxesscommunity.com/viewtopic.php?t=18
